### DISCUSSION TITLE
Attempted break in

### DISCUSSION THREAD

**Discussion by PovilasID** [blakeblackshear/frigate#7063] [Show and tell]
Hello,

So a "alleged potential" burglar was stopped because I got a notification from Home Assonant and Frigate!
 
![trying-to-break-in](https://github.com/blakeblackshear/frigate/assets/396243/6e477d2d-e817-420d-bc08-a03860e7def3)

From this experience I learned some things that mattered to me in this situation  :

1. In events list please move "Download the clip" and "Delete the clip" buttons somewhere else or add a confirm on delete or bulk actions for that list:
Here is visual explanation:
If you want to download a clip you push the blue button
![chrome_EYaUe4pFAc](https://github.com/blakeblackshear/frigate/assets/396243/2ea59b10-f129-4307-b682-d318d07d640b)
It opens a pop up
![chrome_i6rEgn0oZU](https://github.com/blakeblackshear/frigate/assets/396243/9f9d8880-d569-48a9-87b3-cf5e8a0cb9ac)
That covers the delete button on the previous clip which does not need confirmation
![chrome_N0FmJeoOLO](https://github.com/blakeblackshear/frigate/assets/396243/9060c3f2-9382-4205-8105-d2669498d63e)
If you accidently double tap download button you may delete the next clip.

When trying to download clips while on mobile device and mobile connection while in a car in a  state of... 'F**K!'  An accidental inpatient double tap is quite likely. Even after the main event while talking with the police I was not feeling 100% attentive or sure thar I have not accidently deleted a clip. I have a continues record of last 48 hours to fall back on, however if had not this may be much more of an issue.

2. In Frigate's UI there is no option to download recordings.
I was looking to download a recording of full hour to let the investigator view it all and there is an option for it in Home Assistant's Frigates UI plugin but not in Frigate's UI. Also I could not use HA to download hour recording. I need to investigate why but ripping a video from the stream's playback worked so that is fine.

3. Internal Frigate's folder structure does not match the time zone defined in environmental variable. 
Frigate stores recordings in small chunks in folder structure like this `storage/recordings/[date]/[hour-in-24-format]/[camera-name]/[file].mp4` I have backup of Frigate files but after looking at them I realized that [hour-in-24-format] part does not match the correct hour it was in `15` when it should have been in `18` (I am in GMT+3). Maybe Maybe first time I ran Frigate I forgot to define `TZ`. I do not remember by now. However, now I have defined and the issue is still there.  

BONUS: `Send to Frigate+` button in events list is nice way to contribute, however send to _personal cloud_ would be nice. I assume `Send to Frigate+` uploads to some S3 bucket, so most of the code is there. Extra button 'Send to personal cloud'  would be nice way to initiate backup. I learned that then you are on call phone loses internet connectivity, so I can not download a clip on my phone while on a call. However uploading to S3 would not be interrupted. A user would need is to define their creds to their S3 bucket in the config.

The alleged burglar was caught trying to run away. He did not get in but my doors will need some professional TLC. 
If the investigator is up for it I will grant access to him/her directly to Frigate and may share their feedback.

**Comment by NickM-27** [comment]
> 2. In Frigate's UI there is no option to download recordings.
I was looking to download a recording of full hour to let the investigator view it all and there is an option for it in Home Assistant's Frigates UI plugin but not in Frigate's UI. Also I could not use HA to download hour recording. I need to investigate why but ripping a video from the stream's playback worked so that is fine.

This has already been added for 0.13

> 3. Internal Frigate's folder structure does not match the time zone defined in environmental variable.
>
> Frigate stores recordings in small chunks in folder structure like this storage/recordings/[date]/[hour-in-24-format]/[camera-name]/[file].mp4 I have backup of Frigate files but after looking at them I realized that [hour-in-24-format] part does not match the correct hour it was in 15 when it should have been in 18 (I am in GMT+3). Maybe Maybe first time I ran Frigate I forgot to define TZ. I do not remember by now. However, now I have defined and the issue is still there.

This is by design and covered in the docs, frigate recordings are stored in UTC and also not meant to be interacted with by the user directly in the file system (that is what the UI and API are for)

**Reply by NickM-27** [reply]
as for 1:
there are many different perspectives on this, many users don't want confirmation on the delete button so that events can be easily deleted without having to move the mouse. There are many open and pinned tickets for improving the delete ux

**Reply by PovilasID** [reply]
> as for 1: there are many different perspectives on this, many users don't want confirmation on the delete button so that events can be easily deleted without having to move the mouse. There are many open and pinned tickets for improving the delete ux

I think the position might not be such a big issue if there was an option to perform bulk download and delete actions. I get why having delete button same place without confirmation is nice because when a bunch of irrelevant events can be quickly dismissed without moving the mouse. However, again this could be solved by bulk edit. 
What about instead of doing pup up just adding a list of icon buttons and their explanations relegating to hover overs? Just as a temp fix?

> This has already been added for 0.13

Cool!

> This is by design and covered in the docs, frigate recordings are stored in UTC and also not meant to be interacted with by the user directly in the file system (that is what the UI and API are for)

Yah I only was looking at raw files because UI failed me aka now recording download button. I understand how this could make it simpler, however, when trying to backup only key data this will cause some confusion.
